devito/dle/transformer.py

```diff
@@ -438,6 +481,71 @@ def heuristic(dim_size):
 
         return {'nodes': processed, 'arguments': arguments}
 
+    @dle_transformation
+    def _ompize(self, state, **kwargs):
+        """
+        Add OpenMP pragmas to the Iteration/Expression tree to emit parallel code
+        """
+
+        processed = []
+        for node in state.nodes:
+
+            # Handle denormals
+            denormals = FindNodeType(Denormals).visit(state.nodes)
+            mapper = {i: Denormals(header=omplang['par-region']) for i in denormals}
+
+            # Handle parallelizable loops
+            for tree in retrieve_iteration_tree(node):
+                # Note: a 'blocked' Iteration is guaranteed to be 'parallel' too
+                blocked = [i for i in tree if 'blocked' in i.properties]
+                parallelizable = [i for i in tree if 'parallel' in i.properties]
+                candidates = blocked or parallelizable
+                if not candidates:
+                    continue
+
+                # Heuristic: if at least two parallel loops are available and the
+                # physical core count is greater than self.thresholds['collapse'],
+                # then omp-collapse the loops
+                if psutil.cpu_count(logical=False) < self.thresholds['collapse'] or\
+                        len(candidates) < 2:
+                    n = candidates[0]
+                    mapper[n] = List(header=omplang['par-for'], body=n)
+                else:
+                    nodes = candidates[:2]
+                    mapper.update({n: List(header=omplang['par-for-collapse2'], body=n)
```

Comment:
But I am talking about the case where we have more than 4 loops in total (the `StencilKernel` does not assume any maximum number of loops so it could generate a 10 loop nest in theory). 
Even though the innermost loop is being vectorised, we might have more loops available to be parallelised and collapsed. 

So I believe that the `StencilKernel` does not treat the 4 loop case as special. I am asking whether it is the same with this bit of code as well?